### ISSUE TITLE
Implement an inventory

### ISSUE THREAD

**Issue by cbcerquiaga** [cbcerquiaga/YouMeAndTheEndOfTheWorld#8]
Each player should have a personal inventory. How do we want to do that? Should items have size and weight, or just weight? Should we limit certain types of items and not others by that (e.g. in Fallout ammo has no weight)? Should we cap the number or way items are stored (like, you only get x number of weapons)? How should we segment the inventory?

**Comment by UngratefulFool** [comment]
Item should be an object, it will have values for weigh, name, image, worth, description. Inventory should be an object, it will have an array to store items in it. Inventory can have an total weight assigned to it, then there can be a check each time a player picks up an item to see if they have the ability to.

**Comment by cbcerquiaga** [comment]
Right, but the issue is, do we have one type of object for Item with a ton of traits on it (boolean canYouFishWithIt, int calorieVal, string WeaponType) or do we have subclasses for certain types of items, like survival aid (toilet paper, soap, canned food, water), survival tools (tent, matches, backpack), minigame specific items (fishing rod, tennis racquet), wearable items like clothes, crafting scrap (sticks, rope, tape), weapons (maybe even separate classes for melee and ranged ones), and other stuff? And do we have one Inventory that we separate into groups when the player looks at it, or do we have more than one and tally up the weight?
I think that having items have size and weight might make for some interesting player choices later on too, and maybe even cool perks. You'd have to keep track of both how much you can carry and how much it weighs, and maybe the inventory actually fills up when you hit the size limit, and if you hit the weight limit you move slower and more time passes when you leave an area, with a much higher actual weight limit for what you can carry. There could be separate perks for both- like "pack rat" reducing the size of every item by 1, or "rucksack runner" that increases the weight before you slow down. Just a thought, it wouldn't be too hard to implement it one way and switch it if we hate it, as long as we switch before we start filling the game with items.

**Comment by UngratefulFool** [comment]
Completed, very basic, but an inventory. Since we will be adding to the inventory, I don't know if I should close this, because I will continue to keep adding features to it and new items, and better information about your inventory.

**Comment by cbcerquiaga** [comment]
I think we can close it and add new issues for each new task for improving the inventory, just so it's clearer what we're doing at a given time and how much progress we're making.
